cli/azd/pkg/apphost/aca_ingress.go

```diff
@@ -0,0 +1,172 @@
+package apphost
+
+import (
+	"fmt"
+)
+
+// buildAcaIngress builds the Azure Container Apps ingress configuration from the provided bindings.
+func buildAcaIngress(bindings map[string]*Binding, defaultIngressPort int) (*genContainerAppIngress, error) {
+	if len(bindings) == 0 {
+		return nil, nil
+	}
+
+	if err := validateInput(bindings); err != nil {
+		return nil, err
+	}
+
+	endpointByContainerPort := make(map[string][]*Binding)
+	for _, binding := range bindings {
+		bindingKey := "default" // default is for those with no container port
+		if binding.TargetPort != nil {
+			bindingKey = fmt.Sprintf("%d", *binding.TargetPort)
+		}
+		endpointByContainerPort[bindingKey] = append(endpointByContainerPort[bindingKey], binding)
+	}
+
+	endpointByContainerPortProperties := make(map[string]*endpointGroupProperties)
+	for containerPort, bindings := range endpointByContainerPort {
+		props := &endpointGroupProperties{httpOnly: true}
+		for _, binding := range bindings {
+			if binding.TargetPort != nil {
+				props.port = *binding.TargetPort
+			}
+			if binding.External {
+				props.external = true
+			}
+			if binding.Scheme != acaIngressSchemaHttp && binding.Scheme != acaIngressSchemaHttps {
```

Comment:
Just wondering - is it an error for one of the bindings (for the same port) to have an scheme of `tcp` while another has `http` or `https`?  I think that it is an in practice such a manifest would never exist, but I am wondering if we wanted to have some validation about that?

Reply:
Yes it is, I added an error case for this. We can always allow it later but detection isn't bad here.